### PR TITLE
CA: Debugging snapshotter locking optimisation for better transactions

### PR DESCRIPTION
**Pull request by jayantjain93**
#### Which component this PR applies to?
Cluster autoscaler

#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/debuggingsnapshot/debugging_snapshotter.go [hunk 1]
```diff
@@ -172,6 +172,13 @@ func (d *DebuggingSnapshotterImpl) ResponseHandler(w http.ResponseWriter, r *htt
 func (d *DebuggingSnapshotterImpl) IsDataCollectionAllowed() bool {
 	d.Mutex.Lock()
 	defer d.Mutex.Unlock()
+	return d.IsDataCollectionAllowedNoLock()
+}
+
+// IsDataCollectionAllowedNoLock encapsulated the check to know if data collection is currently active
+// The need for NoLock implementation is for cases when the caller funcs have procured the lock
+// for a single transactional execution
+func (d *DebuggingSnapshotterImpl) IsDataCollectionAllowedNoLock() bool {
```
Comment (x13n) [inline cluster-autoscaler/debuggingsnapshot/debugging_snapshotter.go:181]: I don't think the NoLock version should be public: it should only be used internally by the snapshotter.

Comment (jayantjain93) [reply cluster-autoscaler/debuggingsnapshot/debugging_snapshotter.go:181]: This isn't part of the interface, which is public.
However, CP snapshotter extensions should have access to this function, which is what we achieve from it being public method.

**Pull request merged**